### PR TITLE
Update dev instructions

### LINKED ISSUE #1096: HNC: Developer Docs Improvement
I faced some issues following the Developer doc.

1. `source devenv` doesn't work.
```
╭─mbtamuli@techfreak ~/workspace/personal/multi-tenancy/incubator/hnc  ‹master› 
╰─➤  source devenv 
Make sure you source this file, don't run it!
Setting KUBECONFIG to kind cluster
ERROR: unknown flag: --name
╭─mbtamuli@techfreak ~/workspace/personal/multi-tenancy/incubator/hnc  ‹master› 
╰─➤  cat devenv
#!/bin/bash
echo "Make sure you source this file, don't run it!"
echo "Setting KUBECONFIG to kind cluster"
export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
```

2. `CONFIG=kind make deploy` doesn't work, but I already see an open issue about it #1084

Also, not sure if it was just me, but the jump from the section - **Building and deploying to a test cluster** to **Developing with KIND** and then again reading the previous section **Development Workflow** was confusing a tad bit. Not the content, but the sequence of the sections. 

Thoughts? These issues might not be super important, but for a first time contributor it could helpful.

### PR DESCRIPTION
**Pull request by adrianludwin**
Fixes #1096

### CODE REVIEW

**Comment by adrianludwin** [comment]
/assign @yiqigao217 
/assign @rjbez17 

The two commenters on #1096 are happy with this, does it lgty?

/cc @GinnyJI

File: incubator/hnc/README.md [hunk 1]
```diff
@@ -117,60 +117,111 @@ is configured to use the project containing the registry you want to use, and
...
+      special-purpose directories in `test/`. These are basically glorified
+      `bash` scripts with better error checking, and directly invoke `kubectl`.
+  - Ensure `make test` passes - this runs unit tests only.
+  - Deploy to your cluster with `make deploy`
+  - Test your changes by hand and verify that your changes are working
+    correctly. Some ways you can do that are:
+    - Look at logging with `make deploy-watch`
+    - Look at the result of the structure of your namespaces with `kubectl-hns tree -A` or `kubectl-hns tree NAMESPACE`
+    - See the resultant conditions or labels on namespaces by using `kubectl describe namespace NAMESPACE`
+  - Run the e2e tests via `make test-e2e`. This will take about 10-20m, so go
```
Comment (yiqigao217) [inline incubator/hnc/README.md:141]: Maybe add that you will have to have the deployment ready.

**Review by yiqigao217 — COMMENTED** [review]
lgtm just 1 nit.

File: incubator/hnc/README.md (hunk 1 above)
Comment (adrianludwin) [reply incubator/hnc/README.md:141]: We already say that at line 135, or do you think we need to say something more explicit?

Comment (yiqigao217) [reply incubator/hnc/README.md:141]: Not sure people will take these manual or e2e tests alternatively.

Comment (adrianludwin) [reply incubator/hnc/README.md:141]: Ok added a specific note and also mentioned `HNC_REPAIR`

**Review by yiqigao217 — COMMENTED** [review]
/lgtm

**Pull request merged**